src/components/tl/mlx5/tl_mlx5_rcache.c

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "tl_mlx5.h"
+
+static ucs_status_t
+rcache_reg_mr(void *context, ucs_rcache_t *rcache, //NOLINT: rcache is unused
+              void *arg, ucc_rcache_region_t *rregion,
+              uint16_t flags) //NOLINT: flags is unused
+{
+    ucc_tl_mlx5_context_t *ctx      = (ucc_tl_mlx5_context_t *)context;
+    void *                 addr     = (void *)rregion->super.start;
+    ucc_tl_mlx5_reg_t *    mlx5_reg = ucc_tl_mlx5_get_rcache_reg_data(rregion);
+    size_t                 length   = (size_t)(rregion->super.end
+                                                       - rregion->super.start);
+    int *                  change_flag = (int *)arg;
+
+    mlx5_reg->region = rregion;
+    *change_flag     = 1;
+    mlx5_reg->mr     = ibv_reg_mr(ctx->shared_pd, addr, length,
+                              IBV_ACCESS_LOCAL_WRITE | IBV_ACCESS_REMOTE_WRITE);
+    if (!mlx5_reg->mr) {
+        tl_error(ctx->super.super.lib, "failed to register memory");
+        return UCS_ERR_NO_MESSAGE;
+    }
+    return UCS_OK;
+}
+
+static void rcache_dereg_mr(void *        context, //NOLINT: context is unused
+                            ucc_rcache_t *rcache,  //NOLINT: rcache is unused
+                            ucc_rcache_region_t *rregion)
+{
+    ucc_tl_mlx5_reg_t *mlx5_reg = ucc_tl_mlx5_get_rcache_reg_data(rregion);
+
+    ucc_assert(mlx5_reg->region == rregion);
+    ibv_dereg_mr(mlx5_reg->mr);
+    mlx5_reg->mr = NULL;
+}
+
+ucc_status_t tl_mlx5_rcachetl_mlx5_create_rcache(ucc_tl_mlx5_context_t *ctx)
```

Comment:
I think you made a mistake with the name of the function? there is duplicate of  "tl_mlx5_..."

Reply:
yes sorry, fixed